src/crypto/siphash.h

```diff
@@ -44,5 +44,6 @@ class CSipHasher
  */
 uint64_t SipHashUint256(uint64_t k0, uint64_t k1, const uint256& val);
 uint64_t SipHashUint256Extra(uint64_t k0, uint64_t k1, const uint256& val, uint32_t extra);
+uint64_t ModifiedRapidHash(uint64_t k0, uint64_t k1, const uint256& val, uint32_t extra);
```

Comment:
I have also measured this being a bottleneck very often (https://github.com/bitcoin/bitcoin/pull/30442), and @sipa mentioned originally that we can always swap it out with something better later: https://github.com/bitcoin/bitcoin/pull/8020#issuecomment-219569508

Do we need to account for collision attacks here, given that inclusion into the map is not for free?

Reply:
I think collision attacks can't be an issue because the hash implementation still uses the random seed, and the hash itself is of good quality.

Reply:
@sipa thanks, you are of course right; I should have read a bit more about that before... Nevertheless I think it's at least interesting to see what difference a fast hash would make.